src/c2goto/headers/ubuntu20.04/kernel_5.15.0-76/include/asm/uaccess.h

```diff
@@ -0,0 +1,14 @@
+
+#define PAGE_SIZE 4096 // page size
+#define USER_MEMORY_SPACE 10000 //only for simulate the user memory
+#define KERNEL_MEMORY_SPACE 10000// only for simulate the kernel memory
+
+char user_memory[USER_MEMORY_SPACE]; //mock user memory
+char kernel_memory[KERNEL_MEMORY_SPACE];//mock user memory
```

Comment:
The definitions of these two arrays should not be in the header.

Reply:
Hi, if I remove these two arrays , then I think i also need to change some assertion in my function body. May i got more suggestions?  So my goal is to simulate the situations that the copy from user to kernel, or vice versa, may cause corruption illegal memory space

Reply:
And also, I am really thankful for your suggestions:)

Reply:
I meant that the header should just declare the arrays, not define them. In C each *definition* (like the arrays above) results in a new symbol being created, which you probably don't want: the regression tests should refer to *the exact same* arrays the implementation of copy_to/from_user do. To achieve that, C has a distinction between declaration and definition. Symbols should be defined just once (in a translation unit, aka a .c file), while they can be declared multiple times. C headers usually don't contain definitions of such symbols, they just declare that they exist and that they are defined somewhere else. A declaration of an array of 42 chars looks like this:
```c
extern char my_array[42];
```

Reply:
hi，I have fixed all the issues we discussed, and pushed them. May I have a check on those changes to see if there is further issues? thanks !